docs/Writing-New-Builtin-Module.md

```diff
@@ -0,0 +1,166 @@
+Writing New Builtin Module
+==========================
+
+This document provides a guide on how to write a builtin module for IoT.js.
+
+Contents:
+
+* Step-by-Step Guide
+  - Writing Builtin JavaScript Module
+  - Writing Builtin Native Module
```

Comment:
I would put Builtin at the end: Native Module Builtin. It feels more natural to me, but if you wish to keep the current one I don't mind.